src/Autofac.Integration.AspNetCore.Multitenant/AutofacMultitenantServiceProviderFactory.cs

```diff
@@ -83,19 +83,17 @@ public IServiceProvider CreateServiceProvider(ContainerBuilder containerBuilder)
         {
             if (containerBuilder == null) throw new ArgumentNullException(nameof(containerBuilder));
 
-            AutofacMultitenantServiceProvider provider = null;
+            MultitenantContainer multitenantContainer = null;
 
-            containerBuilder.Register(_ => provider)
-                .As<IServiceProvider>()
-                .ExternallyOwned();
+            containerBuilder.Register(_ => multitenantContainer)
+              .AsSelf()
+              .ExternallyOwned();
```

Comment:
In the notes on the PR it says there were two registrations for `IServiceProvider` and that was somehow causing the challenge. However, if we don't register the service provider ourselves... does the ASP.NET Core framework do that for us? What happens when we return the provider at the end of this method? Where does it go?

Reply:
We do. That was the hacked we used for 3.0.0 of ASP.NET Core to make it possible. Because `builder.Populate()` already registers an `IServiceProvider` [here](https://github.com/autofac/Autofac.Extensions.DependencyInjection/blob/develop/src/Autofac.Extensions.DependencyInjection/AutofacRegistration.cs#L96)